randovania/generator/generator.py

```diff
@@ -333,3 +345,25 @@ def actual_status_update(msg: str) -> None:
             )
 
     return result
+
+
+def force_logical_placement(
+    pickups: list[PickupEntry], game: GameDescription, lpp_config: LogicalPickupPlacementConfiguration
+) -> Requirement:
+    if lpp_config is LogicalPickupPlacementConfiguration.MINIMAL:
+        return game.victory_condition
+
+    add_all_pickups = lpp_config is LogicalPickupPlacementConfiguration.ALL
+    resources = ResourceCollection.with_database(game.resource_database)
+
+    for pickup in pickups:
+        if pickup.pickup_category.hinted_as_major or pickup.pickup_category.is_key or add_all_pickups:
```

Comment:
We do have quite a few definitions of "major pickup". Maybe the one that we should be using here is same used for major/minor location split?